hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/transport/server/ratis/ContainerStateMachine.java

```diff
@@ -689,29 +689,8 @@ public CompletableFuture<?> link(DataStream stream, LogEntryProto entry) {
 
     final KeyValueStreamDataChannel kvStreamDataChannel =
         (KeyValueStreamDataChannel) dataChannel;
-
-    final ContainerCommandRequestProto request =
```

Comment:
should we remove `KeyValueStreamDataChannel#putBlockRequest`, `KeyValueStreamDataChannel#buffers` and related code, if we remove PutBlock here ?
As I understand it, in order to create the `PutBlockReques`, we keep a `KeyValueStreamDataChannel#buffers` in memory, which guarantees that the buffers data will only be written at `close`. If we remove the `PutBlock` here, do we not need it anymore?

Reply:
@xichen01 , thanks for reviewing this!

Yes, the `putBlockRequest` can be removed.  

Unfortunately, we still need the `buffers` since clients still send a `putBlockRequest`.  We need the `buffers` so the request won't be written to the file as data.